cpp/ql/src/experimental/Best Practices/UselessTest.ql

```diff
@@ -0,0 +1,50 @@
+/**
+ * @name Useless Test
+ * @description Find any useless test of kind a==8 && a!=7
+ * @kind problem
+ * @problem.severity warning
+ * @id cpp/uselesstest
+ * @tags reliability
+ *       readability
+ */
+
+import cpp
+
+predicate sameStmt(Expr e1, Expr e2) {
+  e1.getNumChild() = e2.getNumChild() and
+  e1.toString() = e2.toString() and
```

Comment:
You shouldn't be using the value produced by `toString` in your query logic as we don't guarantees any backwards compatibility on the strings produced by `toString` (i.e., we might silently change the format of the string in the future and thus break your query logic).